clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -9195,7 +9195,7 @@ static bool checkOpenMPIterationSpace(
         SemaRef.Diag(CollapseLoopCountExpr->getExprLoc(),
                      diag::note_omp_collapse_ordered_expr)
             << 0 << CollapseLoopCountExpr->getSourceRange();
-      else
+      else if (OrderedLoopCountExpr)
```

Comment:
I think better to fix the logic in line 9189:
```
if ((CollapseLoopCountExpr && OrderedLoopCountExpr) || (!CollapseLoopCountExpr && !OrderedLoopCountExpr))
```

Reply:
The problem with this approach is that both `OrderedLoopCountExpr` and `CollapseLoopCountExpr` will be dereferenced on lines 9192 and 9193 if they're both are `nullptr`s.

Reply:
Yep, right